### DISCUSSION TITLE
[FEATURE REQUEST] Generate docs for pure "click" too

### DISCUSSION THREAD

**Discussion by senpos** [tiangolo/typer-cli#136] [Questions] [answered]
Hi!

Autogenerating docs for CLI app in markdown is absoluetly awesome feature!

I miss it a lot in pure `click`. :( I have a bunch of apps which are not likely to be rewritten with `typer` for now, as they are in production and work fine, but having documentation for them would help a lot.

As I can see, documentation is built mostly from click's internal objects.
Do you have any plans on improving `typer-cli` to support `click` apps too? Or perhaps making a separate CLI for that purpose? 🤔 As it seems to be a bit out of scope of the **typer**-cli. But that would enable this package to be used in many more projects, as click is widely used.

Thanks!

**Comment by tiangolo** [comment, marked as answer]
Thanks for the interest!

Typer-CLI is now integrated into Typer, when you install `typer` you get the `typer` command with everything there.

---

Now, about the feature request, I currently don't really need it, I'm not sure what would be required to support it. But I fear it's something that would be something extra for me to maintain, so I would probably rather not have that feature, at least for now.

---

As Typer-CLI is now part of Typer, the feature request would now belong in the Typer repo, I'm about to archive this repo.

An option you would have would be to extract the CLI code in a script and modify it to take a Click app for your needs, that should be doable, I think.

Then we can see if later this becomes a feature needed by many people and I could reconsider it. :nerd_face:
